Hello react native

MProuts opened the pull request:
Completed [this tutorial](https://facebook.github.io/react-native/docs/integration-with-existing-apps) with some workarounds.

MProuts committed:
Update gitignore, commit automatic changes to xcworkspace files

MProuts committed:
Update modal configuration for iOS 13 so that game is fullscreen

MProuts committed:
Move iOS project files to a directory called 'ios/'

MProuts committed:
Add package.json and install react-native via yarn

MProuts committed:
Add Podfile and install pods

MProuts committed:
WIP -- build error

benzado committed:
Update Podfile, etc. so that React Native can be used

benzado committed:
Allow insecure HTTP to localhost for Metro Bundler!

So that the app can "hot reload" JS while the app is running.

benzado committed:
Add a High Scores React Component

MProuts commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Stage 2: “Hello RN!”](https://trello.com/c/UkFZCF9f/1051-stage-2-hello-rn)

MProuts committed:
Run pod install

MProuts committed:
WIP - try to log score on exit

jpendry commented on ios/NumberTileGame/Base.lproj/Main_iPhone.storyboard:
```diff
@@ -13,30 +18,33 @@
+                            </button>
                         </subviews>
-                        <color key="backgroundColor" white="1" alpha="1" colorSpace="custom" customColorSpace="calibratedWhite"/>
+                        <color key="backgroundColor" red="1" green="1" blue="1" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
```
Are we sure about this?

jpendry replied:
See: https://twitter.com/blowdart/status/1185375804613488640

jpendry reviewed (approved):
👍

MProuts committed:
Get writing/reading to archive working

MProuts committed:
Read/write actual top scores

MProuts committed:
WIP - Understand why this doesn't work

MProuts committed:
Handle case where no high scores exist yet

MProuts committed:
Dismiss RN view when 'exit' button is pressed

It seems like F3HViewController's 'self' isn't accessible in the native
method we're exposing to RN. I.e. This doesn't work when the method is
called by RN:

  [self dismissViewControllerAnimated:YES completion:nil]

However, we _can_ access the AppDelegate from the exposed native method.
So we access the RN root view by making a property on the AppDelegate.

Hat Tip: https://moduscreate.com/blog/leverage-existing-ios-views-react-native-app/

MProuts committed:
Make array maniuplation slightly cleaner

MProuts committed:
Store view controller reference on BridgeModule class

MProuts commented:
I accidentally kept pushing commits to this branch after Jimmy reviewed.  I moved the part 2 commits [here](https://github.com/logcheck/iOS-2048/pull/2), which I will merge given Jimmy's 👍.

I'll open a separate PR for the part 3 commits.
